Mathlib/AlgebraicGeometry/Sites/Representability.lean

```diff
@@ -0,0 +1,207 @@
+noncomputable def yonedaGluedToSheaf :
+    zariskiTopology.yoneda.obj (glueData hf).glued ⟶ F where
+  -- The map is obtained by finding an object of `F((glueData hf).glued)`.
+  val := yonedaEquiv.symm
+  -- This section is obtained from gluing the section corresponding to `f i : Hom(-, X i) ⟶ F`.
+    ((glueData hf).sheafValGluedMk (fun i ↦ yonedaEquiv (f i)) (by
+      intro i j
+      dsimp
+      apply yonedaEquiv.symm.injective
+      rw [yonedaEquiv_naturality, Equiv.symm_apply_apply,
+        FunctorToTypes.map_comp_apply, yonedaEquiv_naturality, yonedaEquiv_naturality,
+        Equiv.symm_apply_apply, ← Functor.map_comp_assoc,
+        Functor.relativelyRepresentable.symmetry_fst, ((hf i).rep.isPullback' (f j)).w]))
+
+@[reassoc (attr := simp)]
+lemma yoneda_toGlued_yonedaGluedToSheaf (i : ι) :
+    yoneda.map (toGlued hf i) ≫ (yonedaGluedToSheaf hf).val = f i := by
+  dsimp [yonedaGluedToSheaf, Sheaf.homEquiv, Functor.FullyFaithful.homEquiv]
+  apply yonedaEquiv.injective
+  rw [yonedaEquiv_apply, yonedaEquiv_apply]
+  dsimp
+  simp only [comp_id]
+  apply GlueData.sheafValGluedMk_val
+
+@[simp]
+lemma yonedaGluedToSheaf_app_toGlued {i : ι}  :
+    (yonedaGluedToSheaf hf).val.app _ (toGlued hf i) = yonedaEquiv (f i) := by
+  rw [← yoneda_toGlued_yonedaGluedToSheaf hf i]
+  rfl
+
+@[simp]
+lemma yonedaGluedToSheaf_app_comp {V U : Scheme.{u}} (γ : V ⟶ U) (α : U ⟶ (glueData hf).glued) :
+  (yonedaGluedToSheaf hf).val.app (op V) (γ ≫ α) =
+    F.val.map γ.op ((yonedaGluedToSheaf hf).val.app (op U) α) :=
+  congr_fun ((yonedaGluedToSheaf hf).val.naturality γ.op) α
+
+instance [Presheaf.IsLocallySurjective Scheme.zariskiTopology (Sigma.desc f)] :
+    Sheaf.IsLocallySurjective (yonedaGluedToSheaf hf) :=
+  Presheaf.isLocallySurjective_of_isLocallySurjective_fac _
+    (show Sigma.desc (fun i ↦ yoneda.map (toGlued hf i)) ≫
+      (yonedaGluedToSheaf hf).val = Sigma.desc f by aesop_cat)
+
+lemma comp_toGlued_eq {U : Scheme} {i j : ι} (a : U ⟶ X i) (b : U ⟶ X j)
+    (h : yoneda.map a ≫ f i = yoneda.map b ≫ f j) :
+    a ≫ toGlued hf i = b ≫ toGlued hf j := by
+  rw [← (hf i).rep.lift'_fst a b h, assoc]
+  conv_rhs => rw [← (hf i).rep.lift'_snd a b h, assoc]
+  congr 1
+  exact ((glueData hf).glue_condition i j).symm.trans (by simp [toGlued])
+
+@[simp]
+lemma glueData_openCover_map : (glueData hf).openCover.map j = toGlued hf j := rfl
+
+instance : Sheaf.IsLocallyInjective (yonedaGluedToSheaf hf) where
+  equalizerSieve_mem := by
+    rintro ⟨U⟩ (α β : U ⟶ _) h
+    replace h : (yonedaGluedToSheaf hf).val.app _ α = (yonedaGluedToSheaf hf).val.app _ β := h
+    have mem := grothendieckTopology_cover (glueData hf).openCover
+    refine GrothendieckTopology.superset_covering _ ?_
+      (zariskiTopology.intersection_covering (zariskiTopology.pullback_stable α mem)
+        (zariskiTopology.pullback_stable β mem))
+    rintro V (γ : _ ⟶ U) ⟨⟨W₁, a, _, ⟨i⟩, fac₁⟩, ⟨W₂, b, _, ⟨j⟩, fac₂⟩⟩
+    change γ ≫ α = γ ≫ β
```

Comment:
Can we avoid the `change`?

Reply:
You know the category theory part of the library better than me, so do objections to follow your advice.